app/models/user.rb

```diff
@@ -0,0 +1,18 @@
+###
+#  User class for authenticating via WebAuth
+###
+class User
+  attr_accessor :webauth
+
+  def initialize(webauth)
+    @webauth = webauth
+  end
+
+  def webauth_user?
+    @webauth.present?
```

Comment:
That's fine -- then I guess I'm wondering why you're using `@webauth` in `User` instead of the `webauth` accessor you already have?